src/jit/emitxarch.cpp

```diff
@@ -2903,14 +2903,20 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = tmpDsc->tdTempNum();
         offset = 0;
     }
+    else if (memBase->OperGet() == GT_LCL_VAR_ADDR)
+    {
+        varNum = memBase->AsLclVarCommon()->GetLclNum();
+        offset = 0;
+    }
 
     // Spill temp numbers are negative and start with -1
     // which also happens to be BAD_VAR_NUM. For this reason
     // we also need to check 'tmpDsc != nullptr' here.
     if (varNum != BAD_VAR_NUM || tmpDsc != nullptr)
     {
         // Is the memory op in the source position?
-        if (src->isContainedLclField() || src->isContainedLclVar() || src->isContainedSpillTemp())
+        if (src->isContainedLclField() || src->isContainedLclVar() || src->isContainedSpillTemp() ||
+            src->isContainedMemoryOp())
```

Comment:
>           src->isContainedMemoryOp( [](start = 1, length = 36)

isContainedMemoryOp() subsumes the rest. Hence this entire condition can be replaced by src->isContainedMemoryOp()

bool isContainedMemoryOp() const
    {
        return (isContained() && isMemoryOp()) || isContainedLclVar() || isContainedSpillTemp();
    }

IsMemoryOp() return true if either isIndir() or isLclField()